bluetooth: h5: Fix memory access error

LuoZhongYao opened the pull request:
Fix memory access error, handle the case where H5_HDR_LEN
is 0 (e.g: ACK packet), high-priority events are processed
using bt_recv_prio, otherwise the assertion in
hci_core.c: hci_event function cannot be passed

Signed-off-by: ZhongYao Luo <LuoZhongYao@gmail.com>

LuoZhongYao committed:
bluetooth: h5: Fix memory access error

According to the context information, the processing of
net_buf_add_u8 within the get_evt_buf function is
redundant and incorrect.

Signed-off-by: ZhongYao Luo <LuoZhongYao@gmail.com>

stephanosio commented on drivers/bluetooth/hci/h5.c:
```diff
@@ -397,6 +397,11 @@ static void h5_process_complete_packet(u8_t *hdr)
 		net_buf_put(&h5.rx_queue, buf);
 		break;
 	case HCI_EVENT_PKT:
+		if (buf->len && bt_hci_evt_is_prio(*((u8_t *)buf->data))) {
```
`net_buf.data` is already defined as `u8_t *`, so this cast should not be necessary.

p.s. it would be cleaner to use `struct bt_hci_evt_hdr` to access the `evt` value.

joerchan commented:
@LuoZhongYao Thank you for the fix. Would you mind splitting it up into the two different commits for the issues you are fixing? One for sending priority events through bt_recv_prio, and one for handling the zero length ack packet?

LuoZhongYao committed:
bluetooth: h5:  Handling the the zero length ack packet

If H5_HDR_LEN is equal to zero, remaining will grow
negatively, the commit fixes the problem.

Signed-off-by: ZhongYao Luo <LuoZhongYao@gmail.com>

LuoZhongYao committed:
bluetooth: h5: sending priority events through bt_recv_pri

By default all events are processed through bt_recv, which results
in lost events and subsys/bluetooth/hci_core.c:hci_event function
assertions fail to pass

Signed-off-by: ZhongYao Luo <LuoZhongYao@gmail.com>

dleach02 commented:
Is there an issue bug associated with this change?

LuoZhongYao commented:
> Is there an issue bug associated with this change?
I did not open an issue, but the problem should be 100%, as long as the driver is used and there is a Bluetooth device that supports H5.

joerchan commented:
@LuoZhongYao When discovering bugs it is very useful to create a bug issue. In there we can add information about what the bug is, possible workarounds, and severity.
This allows us to make informed decisions about whether these issues are critical to include in releases, and other people who find the same bug can cherry pick it into their branch if affected by it.

Could you create a bug issue for this one?
Then add a Fixes: # 1234 to the discription of this PR?
